controllers/addons_controller.go

```diff
@@ -308,3 +325,22 @@ func indexHelmReleaseByOwner(o runtime.Object) []string {
 
 	return []string{owner.Name}
 }
+
+func indexHelmRepoByOwner(o runtime.Object) []string {
+	reconciler.Log.V(1).Info("processing", "helmrepositories.source.toolkit.fluxcd.io", utils.LogJSON(o))
+	hr, ok := o.(*sourcev1.HelmRepository)
+	if !ok {
+		return nil
+	}
+	owner := metav1.GetControllerOf(hr)
+	if owner == nil {
+		return nil
+	}
+	if owner.APIVersion != kraanv1alpha1.GroupVersion.String() || owner.Kind != "AddonsLayer" {
+		return nil
+	}
+	log := ctrl.Log.WithName("hr sync")
+	log.Info("Helm Repository associated with layer", "Layer Name", owner.Name, "HR", fmt.Sprintf("%s/%s", hr.GetNamespace(), hr.GetName()))
```

Comment:
Once we are sure this works as designed we will probably want to raise the verbosity level for this log message (and the similar one in the `indexHelmReleaseByOwner` function) to something higher than standard Debug as well.

Using two different loggers in the same function seems likely to be confusing at some point, but I don't know if that might have been intentional.

Reply:
I wanted to highlight that the messages related to the HelmRelease sync function, you are right both should be same